packages/analytics-core/src/plugins/destination.ts

```diff
@@ -297,21 +313,25 @@ export class Destination implements DestinationPlugin {
   }
 
   fulfillRequest(list: Context[], code: number, message: string) {
-    this.saveEvents();
     list.forEach((context) => context.callback(buildResult(context.event, code, message)));
   }
 
   /**
-   * Saves events to storage
-   * This is called on
-   * 1) new events are added to queue; or
-   * 2) response comes back for a request
+   * update the event storage
    */
-  saveEvents() {
+  async updateEvent(dropEventSet: Set<string>) {
     if (!this.config.storageProvider) {
       return;
     }
-    const events = Array.from(this.queue.map((context) => context.event));
-    void this.config.storageProvider.set(this.storageKey, events);
+    const savedEvent = await this.config.storageProvider.get(this.storageKey);
+    const reTriedEvents: Event[] = [];
+    if (savedEvent) {
+      savedEvent.filter((event) => {
+        if (!dropEventSet.has(event.event_type)) {
```

Comment:
By doing this, won't we drop all events of that event_type? Shouldn't we be filtering by insert ids instead?

Reply:
Good call. Originally use insert_id, but it's optional, so changed to event_type. But this is the right concern, and the insert_id is be assigned in the context plugin, there must be a value in the destination plugin. Will update this.